src/hotspot/share/opto/loopnode.hpp

```diff
@@ -1880,6 +1886,13 @@ class PathFrequency {
   float to(Node* n);
 };
 
+// Interface to transform OpaqueLoopInit and OpaqueLoopStride nodes of a Template Assertion Predicate Expression.
+class TransformStrategyForOpaqueLoopNodes : public StackObj {
```

Comment:
The name could reflect that it is only for template assertion predicates.

Reply:
Maybe put it in the scope of `TemplateAssertionPredicateExpression` class? Not sure about this idea yet, just an idea.

Reply:
Should this not be in `predicates.hpp`, together with its implementations?